rotkehlchen/db/updates.py

```diff
@@ -81,6 +84,64 @@ def _get_remote_info_json(self) -> dict[str, Any]:
 
         return json_data
 
+    def update_rpc_nodes(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        rpc nodes for those versions.
+        """
+        for version in range(local_version + 1, to_version + 1):
+            file_url = RPC_NODES_URL.format(branch=self.branch, version=version)
+            try:
+                updates = query_file(file_url, is_json=True)
+            except RemoteError as e:
+                log.warning(f'Failed to update rpc nodes due to {str(e)}')
+                break
+
+            updated_rpc_nodes = updates.get('rpc_nodes')
+            if updated_rpc_nodes is None:
+                log.error(f'Remote update {file_url} does not contain rpc_nodes key')
+                break
+
+            try:
+                nodes_to_add_tuples = [
+                    (node['name'], node['endpoint'], node['owned'], node['active'], str(FVal(node['weight'])), SupportedBlockchain.deserialize(node['blockchain']).value)  # noqa: E501
+                    for node in updated_rpc_nodes['addition']
+                ]
+                nodes_to_remove_tuples = [
+                    (node['endpoint'], node['blockchain'])
+                    for node in updated_rpc_nodes['removal']
+                ]
+            except (KeyError, DeserializationError) as e:
+                msg = f'missing key {str(e)}' if isinstance(e, KeyError) else str(e)
+                log.error(
+                    f'Unable to update rpc nodes to '
+                    f'{version} {updated_rpc_nodes} due to {msg}',
+                )
+                continue
+
+            log.info(
```

Comment:
No no no no. This is totally wrong. @yabirgb @prettyirrelevant did you even read the issue?

This is to change the **default** nodes. Not the user's nodes. Essentially what `populate_rpc_nodes_in_database()` does by reading `nodes.json`. That `nodes.json` should change.

And as an extra you can check the DB of the user, as you do here and **if and only if** a node is completely missing add it, and **if and only if** a node you want to delete should be deleted. But that should only happen for the last version. We don't need to run through all versions of an rpc node update. It's not a DB schema.